app/models/assignment.rb

```diff
@@ -49,18 +51,32 @@ def visible_expectation_results
 
   def accept_new_submission!(submission)
     transaction do
+      messages.destroy_all
       update! submission_id: submission.id
       update_submissions_count!
       update_last_submission!
     end
   end
 
-  def comments
-    Comment.where(submission_id: submission_id).order('date DESC')
+  def messages
+    Message.where(submission_id: submission_id).order('date DESC')
+  end
+
+  def message!(message_data)
```

Comment:
It is not even a verb. Previously it was named `comment` because it _is_ a verb, now we should still call it `comment` or rename to `ask!`, `send_message!` or even `raise_hand!`.